darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -0,0 +1,440 @@
+        norm=LogNorm(),
+        cbar_kws=colorbar_kws,
+    )
+
+    for _, spine in hmap.spines.items():
+        spine.set_visible(True)
+
+    if unique_ranks.size > 1:
+        jgrid.ax_marg_y.barh(
+            y=unique_ranks,
+            width=Hmap_data.sum(axis=1),
+            align="edge",
+            height=-0.99,
+            facecolor="black",
+            lw=0.5,
+        )
+    else:
+        jgrid.ax_marg_y.axis("off")
+
+    jgrid.ax_marg_x.bar(
+        x=np.arange(xbins),
+        height=Hmap_data.sum(axis=0),
+        facecolor="black",
+        align="edge",
+    )
+
+    # only create 4 x-axis tick marks, for figure clarity
+    n_xlabels = 4
+    # create x ticks that scale with the number of xbins in the 2D histogram
+    xticks = np.linspace(0, xbins, n_xlabels)
+    jgrid.ax_joint.set_xticks(xticks)
+    # for the x tick labels, start at 0 and end with the max time (converted to
+    # an integer)
+    max_time = np.max(end_time_data)
+    if max_time < 1:
+        x_ticklabels = np.around(np.linspace(0, max_time, n_xlabels), decimals=2)
+    elif (max_time > 1) & (max_time < 10):
+        x_ticklabels = np.around(np.linspace(0, max_time, n_xlabels), decimals=1)
+    else:
+        x_ticklabels = np.asarray(
+            np.linspace(0, np.ceil(max_time), n_xlabels), dtype=int
+        )
+
+    jgrid.ax_joint.set_xticklabels(x_ticklabels, minor=False)
+
+    if unique_ranks.size > 12:
+        # only use 6 labels if there are many ranks to label. Might want to
+        # do some checks on how many ranks there are earlier on since any
+        # file with 1000's of ranks is too messy to plot this way.
+        n_ylabels = 6
+        sns_yticks = jgrid.ax_joint.get_yticks()
+        yticks = np.linspace(sns_yticks.min(), sns_yticks.max(), n_ylabels)
+        jgrid.ax_joint.set_yticks(yticks)
+        yticklabels = np.asarray(
+            np.linspace(unique_ranks.min(), unique_ranks.max(), n_ylabels), dtype=int
+        )
+    else:
+        # if there are less than 12 unique ranks, just label each
+        # row with its rank index
+        yticklabels = np.asarray(unique_ranks, dtype=int)
+    jgrid.ax_joint.set_yticklabels(yticklabels, minor=False)
+
+    # remove ticks between heatmap and histograms
+    jgrid.ax_marg_x.tick_params(
+        axis="x", bottom=False, labelbottom=False, top=False, labeltop=False
+    )
+    jgrid.ax_marg_y.tick_params(
+        axis="y", left=False, labelleft=False, right=False, labelright=False
+    )
+    # remove ticks showing the heights of the histograms
+    jgrid.ax_marg_x.tick_params(
+        axis="y", left=False, labelleft=False, right=False, labelright=False
+    )
+    jgrid.ax_marg_y.tick_params(
+        axis="x", bottom=False, labelbottom=False, top=False, labeltop=False
+    )
+
+    # set the dimensions of the figure to 6.5" wide x 4.5" tall
+    jgrid.fig.set_size_inches(6.5, 4.5)
+
+    # set the spacing between subplots
+    hspace, wspace = 0.01, 0.02
```

Comment:
Phil recommended I change the spacing between the bar graphs and the heatmap. Should be able to do that here and double check that colorbar is still spaced appropriately.